File: eMovieFinder/eMovieFinder.Model/SearchObjects/BaseSearchObject.cs
namespace eMovieFinder.Model.SearchObjects
{
    public class BaseSearchObject
    {
        public int? PageNumber { get; set; }
        public int? PageSize { get; set; }
        public string? OrderBy { get; set; }
        public bool? IsDescending { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/searchobjects/base_search_object.dart
import 'package:json_annotation/json_annotation.dart';

part 'base_search_object.g.dart';

@JsonSerializable()
class BaseSearchObject {
  final int? pageNumber;
  final int? pageSize;
  final String? orderBy;
  final bool? isDescending;

  BaseSearchObject({
    this.pageNumber,
    this.pageSize,
    this.orderBy,
    this.isDescending,
  });

  factory BaseSearchObject.fromJson(Map<String, dynamic> json) => _$BaseSearchObjectFromJson(json);
  Map<String, dynamic> toJson() => _$BaseSearchObjectToJson(this);
}